src/confluence/subcommands/merge.py

```diff
@@ -183,17 +181,41 @@ def main(args):
     setup_logging(args.loglevel)
     _logger.debug(f"Starting merge operation...")
     # read input files
-    _logger.debug(f"Reading files: {args.filelist}")
+    _logger.debug(f"Validating files: {args.filelist}")
+    try:
+        validate_files(args.filelist)
+    except ValueError:
+        sys.exit(0)
     data = OrderedDict()
-    for od in [read(fname, index_col=args.index) for fname in args.filelist]:
-        for k,v in od.items():
-            data[k] = data.get(k, []) + [v]
+    for od, fname in [(read(fname, index_col=args.index), fname) for fname in args.filelist]:
+        if isinstance(od, (OrderedDict, dict)):
+            for k,v in od.items():
+                try:
+                    v = validate_dataframe(v)
+                    data[k] = data.get(k, []) + [v]
+                    _logger.info(f"Dataframe from file {fname} in sheetname {k} has passed validating")
+                except ValueError:
+                    raise ValueError(f"Empty or duplicated cell in file '{fname}' "
+                                     f"in sheet '{k}' "
+                                     f"in column '{v.index.name}'")
+        else:
+            try:
+                od = validate_dataframe(od)
+                data['merged'] = data.get('merged', []) + [od]
+            except ValueError:
+                raise ValueError(f"Empty or duplicated cell in file '{fname}' "
+                                 f"in column '{od.index.name}'")
     # merge consecutive data frames
     _logger.debug(f"Joining data using {args.resolve} to "
                   "resolve merge conflicts.")
     for k, v in data.items():
-        data[k] = functools.reduce(
-            lambda lhs, rhs: merge(lhs, rhs, args.resolve), data[k])
+        try:
+            data[k] = reduce(partial(merge, resolution=args.resolve), data[k])
+            data[k] = validate_dataframe(data[k]) if args.validate else data[k]
```

Comment:
*ibid*

It feels like the additional validation steps are too linear, and this function has become too long. Can you split this into functions to simplify `main`?